### ISSUE TITLE
Rule: Catch hyphens usage inside variable names

### ISSUE THREAD

**Issue by mnojek** [MarketSquare/robotframework-robocop#271]
Using hyphens inside variable names can be sometimes misunderstood by the interpreter as minus sign and break the execution. Better to use underscores instead. Would be nice to have a rule that suggests it.

**Comment by mnojek** [comment]
Sometimes tester may need to create variables with hyphens inside the variable name to calculate something in-place. Because of that, it's hard to decide which case should be accepted.

Let's consider `s` variable with `2` value as string:
```
${s}  Set Variable  2
${r}  Set Variable  ${s-1}  # this will give TypeError: unsupported operand type(s) for -: 'str' and 'int'
${r}  Set Variable  ${2-s}  # this will give NameError: name 's' is not defined
```
Also let's try defining variable `i` with value `2` as integer:
```
${i}  Set Variable  ${2}
${r}  Set Variable  ${i-1}  # this will return 1
${r}  Set Variable  ${2-i}  # this will give NameError: name 's' is not defined
```
and let's look also at this:
```
${r}    Set Variable  ${2-1}  # this will return 1
${a-b}  Set Variable  1       # this is possible
```

Things are more complicated than I thought. Some of these examples are detected by interpreter, but some are accepted.
Should we skip adding this rule? I cannot think how we can help people avoid these problems with Robocop. What do you think @bhirsz?

**Comment by bhirsz** [comment]
There is just one alternative - we could accept hyphen in arguments but warn on hyphens in return (assign) values. For example:
```
${r}    Set Variable  ${2-1}  # this is fine
${a-b}  Set Variable  1       # this will warn - because if it's later used as ${a-b} it can lead to ambigious results
```
If tester need to calculate something in place he/she will do it using variable as argument and not assign value.
